tests/util/types.js

```diff
@@ -0,0 +1,135 @@
+const types = {
+	any: {
+		syntax: "*",
+		initialValue: "0",
+		values: ["foo", "42"],
+	},
+
+	integer: {
+		initialValue: "0",
+		values: ["42", "-42"],
+		property: "z-index",
+	},
+
+	number: {
+		initialValue: "0",
+		values: ["42", "42.42", "-0.1", "calc(infinity)"],
+		property: "opacity",
+	},
+
+	length: {
+		initialValue: "0px",
+		values: ["42px", "-42px"],
+		property: "width",
+	},
+
+	percentage: {
+		initialValue: "0%",
+		values: ["42%", "-42%"],
+		property: "opacity", // actually <number> | <percentage> but there is no property that only does <percentage>
+	},
+
+	"length-percentage": {
+		initialValue: "0%",
+		values: ["42px", "-42px", "42%", "-42%"],
+		property: "width",
+	},
+
+	color: {
+		initialValue: "rgba(0, 0, 0, 0)",
+		values: ["rgb(0, 0, 0)", "hsl(0, 100%, 50%)"],
+		property: "color",
+	},
+
+	angle: {
+		initialValue: "0deg",
+		values: ["0.5turn", "360deg"],
+		property: "rotate",
+	},
+
+	resolution: {
+		initialValue: "96dpi",
+		values: ["42dppx", "42dpi"],
+		property: "resolution",
```

Comment:
Isn't that a media feature we can't test without MQ? Setting this property on an element and getting its computed value will always return an empty string.

Reply:
Ah right, that's not an actual property, `image-resolution` is! 🤦🏼‍♀️ https://developer.mozilla.org/en-US/docs/Web/CSS/image-resolution
Not sure if that's supported anywhere. Which reminds me, we need to set the test to `skip` if the property is not supported.

Reply:
It’s not supported anywhere.

What if, for skipping such tests, we do something like `this.skip = !CSS.supports(meta.property, "inherit")`?